### PR TITLE
Auth: Add empty role definition

### PR DESCRIPTION
**Pull request by Jguer**
- Backend only
- Add empty role definition
- Not selectable

---

possible to select from API but it's predicted as we'll build frontend and UX from here

### CODE REVIEW

**Review by mckn — COMMENTED** [review]
LGTM!

File: pkg/models/roletype/role_type.go [hunk 1]
```diff
@@ -9,22 +9,26 @@ import (
...
 	RoleEditor RoleType = "Editor"
 	RoleAdmin  RoleType = "Admin"
 )
 
 func (r RoleType) IsValid() bool {
-	return r == RoleViewer || r == RoleAdmin || r == RoleEditor
+	return r == RoleViewer || r == RoleAdmin || r == RoleEditor || r == RoleNone
 }
 
 func (r RoleType) Includes(other RoleType) bool {
```
Comment (kalleep) [inline pkg/models/roletype/role_type.go:22]: Nit: I think we can simplify this check by using a precedence table for role types:

```go
var rolePrecedence = map[RoleType]int{
	None:   1,
	Viewer: 2,
	Editor: 3,
	Admin:  4,
}

func (r RoleType) Includes(other RoleType) bool {
	return rolePrecedence[r] >= rolePrecedence[other]
}
```

We could also use this table to simplify the other functions.

```go
func (r RoleType) IsValid() bool {
	_, ok := rolePrecedence[r]
	return ok
}

// Needed to keep stable order
var roleOrder = [...]RoleType{
	None,
	Viewer,
	Editor,
	Admin,
}

func (r RoleType) Children() []RoleType {
	children := make([]RoleType, 0, 3)

	for _, role := range roleOrder {
		if rolePrecedence[r] > rolePrecedence[role] {
			children = append(children, role)
		}
	}

	return children
}

func (r RoleType) Parents() []RoleType {
	parents := make([]RoleType, 0, 3)

	for _, role := range roleOrder {
		if rolePrecedence[r] < rolePrecedence[role] {
			parents = append(parents, role)
		}
	}

	return parents
}

```

File: pkg/services/org/model.go [hunk 2]
```diff
@@ -47,6 +47,7 @@ type OrgUser struct {
 type RoleType = roletype.RoleType
 
 const (
+	RoleNone   RoleType = "None"
```
Comment (kalleep) [inline pkg/services/org/model.go:50]:
```suggestion
	RoleNone RoleType = roletype.None
```

File: pkg/services/sqlstore/migrations/ualert/permissions.go [hunk 3]
```diff
@@ -17,13 +17,14 @@ import (
 type roleType string
 
 const (
+	RoleNone   roleType = "None"
```
Comment (kalleep) [inline pkg/services/sqlstore/migrations/ualert/permissions.go:20]: I don't think we need to change this migration right?

**Review by kalleep — APPROVED** [review]
Looks good to me, left some suggestions

File: pkg/services/sqlstore/migrations/ualert/permissions.go (hunk 3 above)
Comment (Jguer) [reply pkg/services/sqlstore/migrations/ualert/permissions.go:20]: normally not, I didn't find a reason to not change it, just in case

**Review by kalleep — APPROVED** [review]

**Review by mgyongyosi — APPROVED** [review]

**Pull request merged**